verification/client/abi.go

```diff
@@ -37,17 +37,67 @@ type Support struct {
 	RetainParentContext bool
 }
 
-// All DPE profile command codes
-const (
-	CommandGetProfile          CommandCode = 0x1
-	CommandInitializeContext   CommandCode = 0x7
-	CommandDeriveContext       CommandCode = 0x8
-	CommandCertifyKey          CommandCode = 0x9
-	CommandSign                CommandCode = 0xa
-	CommandRotateContextHandle CommandCode = 0xe
-	CommandDestroyContext      CommandCode = 0xf
-	CommandGetCertificateChain CommandCode = 0x10
-)
+// profileCommandCodes holds command codes for a specific revision of the
+// DPE iRoT profile.
+type profileCommandCodes struct {
+	GetProfile          CommandCode
+	InitializeContext   CommandCode
+	DeriveContext       CommandCode
+	CertifyKey          CommandCode
+	Sign                CommandCode
+	RotateContextHandle CommandCode
+	DestroyContext      CommandCode
+	GetCertificateChain CommandCode
+	ExtendTCI           CommandCode
+}
+
+// profileInfo holds constants defined in a specific version of the DPE iRoT
+// profile.
+type profileInfo struct {
+	MajorVersion uint16
+	MinorVersion uint16
+	Codes        profileCommandCodes
+}
+
+// getProfileInfoV08 returns profile info for v0.8 of the DPE iRoT profile
+func getProfileInfoV08() profileInfo {
+	return profileInfo{
+		Codes: profileCommandCodes{
+			GetProfile:          0x1,
+			InitializeContext:   0x5,
+			DeriveContext:       0x6,
+			CertifyKey:          0x7,
+			Sign:                0x8,
+			RotateContextHandle: 0xE,
+			DestroyContext:      0xF,
+			GetCertificateChain: 0x80,
+		},
+		MajorVersion: 0,
+		MinorVersion: 8,
+	}
+}
+
+// getProfileInfoV09 returns profile info for v0.9 of the DPE iRoT profile
+func getProfileInfoV09() profileInfo {
+	return profileInfo{
+		Codes: profileCommandCodes{
+			GetProfile:          0x1,
+			InitializeContext:   0x7,
+			DeriveContext:       0x8,
+			CertifyKey:          0x9,
+			Sign:                0xa,
+			RotateContextHandle: 0xe,
+			DestroyContext:      0xf,
+			GetCertificateChain: 0x10,
+		},
+		MajorVersion: 0,
+		MinorVersion: 9,
+	}
+}
+
+func getProfileInfoLatest() profileInfo {
```

Comment:
Maybe add a comment here saying that this function must be updated whenever a new version of the profile is released

Reply:
I just removed this function. I ended up choosing which profileInfo to use based on the version returned by GetProfile.